pkg/config/remote/service/service.go

```diff
@@ -150,19 +153,52 @@ func init() {
 	exportedMapStatus.Set("lastError", &exportedLastUpdateErr)
 }
 
+type configurableDurationOpt struct {
+	interval time.Duration
+	cfgPath  string
+}
+
+type configurableIntOpt struct {
+	value   int
+	cfgPath string
+}
+
 type options struct {
-	traceAgentEnv    string
-	databaseFileName string
+	flavor                 string
+	rcKey                  string
+	apiKey                 string
+	traceAgentEnv          string
+	databaseFileName       string
+	configRootOverride     string
+	directorRootOverride   string
+	clientCacheBypassLimit *configurableIntOpt
+	refresh                *configurableDurationOpt
+	maxBackoff             *configurableDurationOpt
+	clientTTL              *configurableDurationOpt
 }
 
 var defaultOptions = options{
-	traceAgentEnv:    "",
-	databaseFileName: "remote-config.db",
+	flavor:                 "Remote Config",
+	rcKey:                  "",
+	apiKey:                 "",
+	traceAgentEnv:          "",
+	databaseFileName:       "remote-config.db",
+	configRootOverride:     "",
+	directorRootOverride:   "",
+	clientCacheBypassLimit: nil,
+	refresh:                nil,
+	maxBackoff:             nil,
+	clientTTL:              nil,
 }
 
 // Option is a service option
 type Option func(s *options)
 
+// WithFlavor sets the service's flavor, which is used during logging/metrics to distinguish between different instances of the RC service
+func WithFlavor(flavor string) func(s *options) {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ Question:** ‏Shouldn't we make it mandatory?

Reply:
If we do it shouldn't be an Option func and should be part of the constructor explicitly.

Also, small nit: `flavor` has meaning already in the agent IIUC (cluster-agent, trace-agent, etc) so it might be good to not reuse this term for differentiating between HA and non-HA `Service`.

Reply:
Yeah my thinking was that the "normal" RC service would just be the default value but I can make it required as a param. And refactor the name 😄